source/app/schedulers/instance_scheduler.py

```diff
@@ -159,11 +171,53 @@ def get_session_for_account(cross_account_role, aws_account):
                     aws_secret_access_key=credentials["SecretAccessKey"],
                     aws_session_token=credentials["SessionToken"],
                 )
-            except Exception as ex:
+            except ClientError as ex:
                 self._logger.error(
-                    ERR_ASSUMING_ROLE.format(cross_account_role, aws_account, str(ex))
+                    "Error Code {}".format(ex.response.get("Error", {}).get("Code"))
                 )
-                return None
+                if ex.response.get("Error", {}).get("Code") == "AccessDenied":
+                    try:
+                        self._logger.error(
+                            "Removing the account {} from scheduling configuration as assume role permission is missing for the iam role {}".format(
+                                aws_account, cross_account_role
+                            )
```

Comment:
Is this the right thing to do? If the customer reinstalls the spoke stack will be the account be added back?